### PR TITLE
New options for two options

### PR DESCRIPTION
**Pull request by graphemecluster**
Refer to #42.

### CODE REVIEW

File: Cantoboard/SettingViewController.swift [hunk 1]
```diff
@@ -58,10 +58,11 @@ class SettingViewController: UITableViewController, UIGestureRecognizerDelegate
     @objc func updateSettings(_ sender: Any) {
         let selectedSymbolShape: SymbolShape
         switch symbolShapeControl.selectedSegmentIndex {
-        case 0: selectedSymbolShape = .half
-        case 1: selectedSymbolShape = .full
-        case 2: selectedSymbolShape = .smart
-        default: selectedSymbolShape = .smart
+        case 0: selectedSymbolShape = .default
```
Comment (alex-the-man) [inline Cantoboard/SettingViewController.swift:61]: What does .default stand for?

**Comment by alex-the-man** [comment]
If one of the goal is to enable:

>  Option to use buffered input mode in Chinese but immediate input mode in English to match iOS default behavior (New options for two options #61)

I like this idea. I think we could implement it as the default behavior without adding a new mode. However, we should do it in a separate commit, with extensive manual testing to make sure switching between mode doesn't break anything.

**Comment by alex-the-man** [comment]
> Option to determine half/full width according to Chinese/English input mode but not context before to match iOS default behavior

We could add a new mode in a separate commit. Please rename the old smart mode to contextual, and add a new mode: language. We can discuss on the user visible naming for these 2 modes.

**Comment by graphemecluster** [comment]
> I think we could implement it as the default behavior without adding a new mode. 

Do you mean that we can simply remove the whole option?

> However, we should do it in a separate commit, with extensive manual testing to make sure switching between mode doesn't break anything.

> We could add a new mode in a separate commit.

Do you mean I should separate the single commit into 4 commits with a force-push, or opening 4 PRs?

> Please rename the old smart mode to contextual, and add a new mode: language. We can discuss on the user visible naming for these 2 modes.

OK, I will rename `.default` and `.smart` to `.language` and `.contextual` respectively.

**Comment by alex-the-man** [comment]
> Do you mean that we can simply remove the whole option?

Yep. Remove .default u added. Change the behavior of "two stage" to use immediate input in English mode. Remember to test switching between modes. It will probably break.

> Do you mean I should separate the single commit into 4 commits with a force-push, or opening 4 PRs?

Just 2 PRs. One for each option.

**Comment by graphemecluster** [comment]
Opened #63, #64.

**Pull request closed without merging**